File: javascript/01-circles-demo/js/step-02.js

// Step 2: Draw a row of circles using a loop
// This sketch shows how to use a for loop to draw a row of circles

function setup() {
  createCanvas(800, 800); // Create a canvas 800x800 pixels
  background(220); // Light gray background
  noStroke(); // No outline on circles
  fill(100, 150, 255); // Set fill color
  let numCircles = 8; // Number of circles in the row
  let spacing = width / (numCircles + 1); // Even spacing between circles
  for (let i = 1; i <= numCircles; i++) {
    ellipse(i * spacing, height/2, 80); // Draw each circle in the row
  }
}

// No animation in this step

File: python/01-circles-demo/py/step-02.py
# Demo 1: Circle Grid with Randomness (Processing.py)
# Step 2: Draw a grid of circles using loops
# This is the Python (Processing.py) version of the p5.js step-01 demo.
# For use in Processing IDE, Python mode.

def setup():
    size(600, 600)  # Set the canvas size (width, height)
    background(255) # White background
    noStroke()      # No outline for circles
    fill(80, 180, 255)  # Light blue fill
    cols = 10       # Number of columns
    rows = 10       # Number of rows
    diameter = 40   # Circle diameter
    padding = 20    # Space between circles
    for y in range(rows):
        for x in range(cols):
            cx = padding + x * (diameter + padding)
            cy = padding + y * (diameter + padding)
            ellipse(cx, cy, diameter, diameter)
